### ISSUE TITLE
Does not compile with markdown3

### ISSUE THREAD

**Issue by Esokrates** [pdfpc/pdfpc#682]
**Expected Behaviour**

Project should build successfully using latest vala version.

**Actual Behaviour**

Compilation fails with message

```
pdfpc-4.6.0/build/src/classes/renderer/markdown.c:191:20: error: variable or field 'flags' declared void
```
see the following build log against Opensuse Tumbleweed:
https://build.opensuse.org/build/home:esokrates/openSUSE_Tumbleweed/x86_64/pdfpc/_log

**Steps to reproduce**

1. Try to build version 4.5.0 or 4.6.0 on Opensuse Tumbleweed.
2. Observe that the build fails.

pdfpc version: 4.6.0
Used Distribution (GTK version, Vala version): Opensuse Tumbleweed, vala 0.56.5
Display Server (X11/Wayland): does not matter (build error)
HiDPI Setting (yes/no): does not matter (build error)

**Comment by Esokrates** [comment]
Possibly https://gitlab.gnome.org/GNOME/vala/-/issues/1424

**Comment by fnevgeny** [comment]
Can you show what is at line 191 of pdfpc-4.6.0/build/src/classes/renderer/markdown.c?

**Comment by Orc** [comment]
Yes, I changed the flags format because I ran out of bits in the old version.  already compiled code should work with the new libraries (I wedged a v2compat layer in) but new code needs to use the new flags structure.   I apologize for it not actually being *documented*;  I've been transitioning and between it and the previous wave of dysphoria I've been running at about 1/10th of my abilities for the last couple of years.)

The tl/dr; of the new flags is that flags are now a `mkd_flag_t*`, which need to be allocated with the function `mkd_flags()` and, when you're done, deallocated with `mkd_free_flags(mkd_flag_t*)`, like so:


~~~~
main()
{
    mkd_flags_t *flags = mkd_flags();

    if ( flags == NULL ) { perror(errno); exit(1); }

    /* ... do the usual; markdown stuff; all the mkd_ functions that previously
        expected a mkd_flag_t flags argument  now expect a mkd_flag_t pointer,
        so if the flags are allocated like this they will just fall into the existing
        code  */

    mkd_free_flags(flags);
}
~~~~

**Comment by wentasah** [comment]
Possible fix is in #716. @Orc thanks for the hint what needed to be changed.
